lib/Dialect/ESI/runtime/cpp/tools/esiquery.cpp

```diff
@@ -36,12 +39,40 @@ int main(int argc, const char *argv[]) {
   if (argc > 3)
     cmd = argv[3];
 
-  std::unique_ptr<Accelerator> acc = Accelerator::connect(backend, conn);
-  const SysInfo &info = acc->sysInfo();
+  try {
+    std::unique_ptr<Accelerator> acc = registry::connect(backend, conn);
+    const auto &info = *acc->getService<services::SysInfo>();
+
+    if (cmd == "version")
+      std::cout << "ESI system version: " << info.esiVersion() << std::endl;
+    else if (cmd == "json_manifest")
+      std::cout << info.jsonManifest() << std::endl;
+    else if (cmd == "info")
+      printInfo(std::cout, *acc);
+    // TODO: add a command to print out the instance hierarchy.
+    else
+      std::cout << "Connection successful." << std::endl;
+
+    if (cmd.empty())
+      return 0;
+    std::cerr << "Unknown command: " << cmd << std::endl;
+    return 1;
 
-  // Only support the 'version' command.
-  if (cmd == "version")
-    std::cout << "ESI system version: " << info.esiVersion() << std::endl;
+  } catch (std::exception &e) {
+    std::cerr << "Error: " << e.what() << std::endl;
+    return -1;
+  }
+}
 
-  return 0;
+void printInfo(std::ostream &os, Accelerator &acc) {
+  std::string jsonManifest =
+      acc.getService<services::SysInfo>()->jsonManifest();
+  Manifest m(jsonManifest);
+  os << "API version: " << m.apiVersion() << std::endl << std::endl;
+  os << "********************************" << std::endl;
+  os << "* Design information" << std::endl;
+  os << "********************************" << std::endl;
```

Comment:
Nit: 80-character wide `*`s.

Reply:
I actually don't like such wide comments in my terminal. I think it makes it look like I've run two commands.